txnkv/txnsnapshot/snapshot.go

```diff
@@ -391,6 +391,7 @@ func (s *KVSnapshot) batchGetSingleRegion(bo *retry.Backoffer, batch batchKeys,
 			Keys:    pending,
 			Version: s.version,
 		}, s.mu.replicaRead, &s.replicaReadSeed, kvrpcpb.Context{
+			StartTs:           s.version,
```

Comment:
all the places that you set `StartTs` is just a copy of start version in each type of command. How about getting the start ts with a match of the request type instead of adding a new field in context